src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +248,43 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the extension of the file name of this path as a {@code String},
+     * where the extension is defined to be the portion of the file name string
+     * after the last dot ('.').  If the first character in the name is a dot it
+     * is ignored.  If the extension cannot be determined, then the empty string
+     * is returned.  This will occur if the file name has fewer than three
+     * characters, does not contain a dot, only the first character is a dot,
+     * or the last character is a dot.
+     *
```

Comment:
Thanks for bringing this one back. The survey and the current proposal seems reasonable.

The spec will need cover the case that the Path doesn't have a file name (getFileName can return null so I assume the prototype implementation will NPE in that case). Are you planning to return null or the empty String for this case?

"file name string" needs to be replaced with "the String representation of the file name". This is significant aspect of the proposal in that it aligns the method with toString rather than other operations that preserve the platform representation of the path.

Reply:
The `getFileName() == null` case was an oversight. I think the empty string should be returned in this case.

Reply:
Addressed by commit 8d462f8615bc9038a4e27ec5302902f71d558e02.